src/test/java/org/broadinstitute/hellbender/utils/read/GATKReadAdaptersUnitTest.java

```diff
@@ -1394,4 +1394,59 @@ private int parseSAMStringFlags(String samString) {
         Assert.assertTrue(m.find());
         return Integer.parseInt(m.group(2)); // use the second capture group in the reg ex above since we don't want the tab
     }
+
+    @DataProvider(name="copyData")
+    public Object[][] getCopyData() {
+        List<Object[]> testCases = new ArrayList<>();
+
+        testCases.add(new Object[]{basicReadBackedBySam()});
+        testCases.add(new Object[]{basicReadBackedByGoogle()});
```

Comment:
That pulls on a long thread, and is the reason I asked yesterday about the status of ADAM backed reads. It currently doesn't have copy or equals implementations, so copy always results in a SAMRecord backed read; the records are never equal to themselves; it never updates its backing AlignmentRecord. I started down the path of fixing those and ran into problems with the constructors and converters. So I think these things should be part of #1189, but let me know if you want it to be part of this.